### PR TITLE
tinyproxy: 1.10.0 -> 1.11.0

### PR DESCRIPTION
**Pull request by centromere**
###### Motivation for this change

Upgrade to latest version.

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122757` at 99979d20 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>tinyproxy</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/networking/tinyproxy/default.nix:33:5:
```
   |
33 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/tools/networking/tinyproxy/default.nix [hunk 1]
```diff
@@ -1,55 +1,36 @@
...
-          'LDFLAGS="-Wl, $LDFLAGS"'
-  '';
+  configureFlags =
+    optionals    (withDebug)           [ "--enable-debug" ]             # Enable debugging support code and methods.
+    ++ optionals (!withXTinyProxy)     [ "--disable-xtinyproxy" ]       # Compile in support for the XTinyproxy header, which is sent to any web server in your domain.
+    ++ optionals (!withFilter)         [ "--disable-filter"]            # Allows Tinyproxy to filter out certain domains and URLs.
+    ++ optionals (!withUpstream)       [ "--disable-upstream" ]         # Enable support for proxying connections through another proxy server.
+    ++ optionals (!withTransparent)    [ "--disable-transparent" ]      # Allow Tinyproxy to be used as a transparent proxy daemon.
+    ++ optionals (!withReverse)        [ "--disable-reverse" ]          # Enable reverse proxying.
+    ++ optionals (!withManpageSupport) [ "--disable-manpage_support" ]; # Enable support for building manpages.
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/tinyproxy/default.nix:28]: Can we just enable all options? That would make the package way less complicated.

Comment (centromere) [reply pkgs/tools/networking/tinyproxy/default.nix:28]: Would you agree to a toggle just for debugging mode?

Comment (SuperSandro2000) [reply pkgs/tools/networking/tinyproxy/default.nix:28]: If that is usefull, we can do that.

File: pkgs/tools/networking/tinyproxy/default.nix [hunk 2]
```diff
@@ -1,55 +1,26 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, asciidoc, libxml2,
-  libxslt, docbook_xsl }:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, perl, withDebug ? false }:
 
+with lib;
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/tinyproxy/default.nix:4]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/networking/tinyproxy/default.nix [hunk 3]
```diff
@@ -1,55 +1,26 @@
...
-    "--enable-filter"      # Allows Tinyproxy to filter out certain domains and URLs.
-    "--enable-upstream"    # Enable support for proxying connections through another proxy server.
-    "--enable-transparent" # Allow Tinyproxy to be used as a transparent proxy daemon.
-    "--enable-reverse"     # Enable reverse proxying.
-  ] ++
-  # See: https://github.com/tinyproxy/tinyproxy/issues/1
-  lib.optional stdenv.isDarwin "--disable-regexcheck";
-
-  meta = with lib; {
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/tinyproxy/default.nix:20]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by mweinelt** [comment]
@GrahamcOfBorg eval

**Pull request merged**